src/inthost.rs

```diff
@@ -186,30 +202,124 @@ pub(crate) fn gen_remove_actor(msg: Vec<u8>, binding: &str, capid: &str) -> Invo
 #[derive(Debug, Clone)]
 #[cfg_attr(feature = "lattice", derive(serde::Serialize, serde::Deserialize))]
 pub struct Invocation {
-    pub origin: String,
-    pub target: InvocationTarget,
+    pub origin: WasccEntity,
+    pub target: WasccEntity,
     pub operation: String,
     pub msg: Vec<u8>,
     pub id: String,
+    pub encoded_claims: String,
+    pub host_id: String,
 }
 
 /// Represents an invocation target - either an actor or a bound capability provider
 #[derive(Debug, Clone, PartialEq)]
 #[cfg_attr(feature = "lattice", derive(serde::Serialize, serde::Deserialize))]
-pub enum InvocationTarget {
+pub enum WasccEntity {
     Actor(String),
     Capability { capid: String, binding: String },
 }
 
+impl WasccEntity {
+    pub fn url(&self) -> String {
+        match self {
+            WasccEntity::Actor(pk) => format!("{}://{}", bus::URL_SCHEME, pk),
+            WasccEntity::Capability { capid, binding } => format!(
+                "{}://{}/{}",
+                bus::URL_SCHEME,
+                capid.replace(":", "/").replace(" ", "_").to_lowercase(),
```

Comment:
Can use `normalize_capid` from `src/bus/mod.rs`?

Reply:
The bus normalize converts the colons into dots, and this conversion makes them slashes and more url-like.

Reply:
OK.